tests/bluetooth/tester/src/audio/btp_bap_broadcast.c

```diff
@@ -52,12 +54,67 @@ static inline struct bt_bap_stream *stream_broadcast_to_bap(struct btp_bap_broad
 	return &stream->audio_stream.cap_stream.bap_stream;
 }
 
-struct btp_bap_broadcast_local_source *btp_bap_broadcast_local_source_get(uint8_t source_id)
+struct btp_bap_broadcast_local_source *btp_bap_broadcast_local_source_allocate(
+	uint32_t broadcast_id)
 {
-	/* Only one local broadcast source supported for now */
-	(void) source_id;
+	for (uint8_t i = 0; i < ARRAY_SIZE(local_sources); i++) {
+		if (local_sources[i].broadcast_id == broadcast_id) {
+			LOG_ERR("Local source already allocated for broadcast id %d", broadcast_id);
 
-	return &local_source;
+			return NULL;
+		}
+	}
+
+	for (uint8_t i = 0; i < ARRAY_SIZE(local_sources); i++) {
+		if (!local_sources[i].allocated) {
+			local_sources[i].allocated = true;
+			local_sources[i].source_id = i;
+			local_sources[i].broadcast_id = broadcast_id;
+
+			return &local_sources[i];
+		}
+	}
+
+	return NULL;
+}
+
+static int btp_bap_broadcast_local_source_free(struct btp_bap_broadcast_local_source *source)
+{
+	if (source == NULL) {
+		return -EINVAL;
+	}
+
+	memset(&local_sources[source->source_id], 0, sizeof(*source));
+
+	return 0;
+}
+
+struct btp_bap_broadcast_local_source *btp_bap_broadcast_local_source_from_src_id_get(
+	uint32_t source_id)
+{
+	for (uint8_t i = 0; i < ARRAY_SIZE(local_sources); i++) {
+		if (local_sources[i].source_id == source_id) {
+			return &local_sources[i];
+		}
+	}
+
+	LOG_ERR("No local source found with source id %u", source_id);
+
+	return NULL;
+}
+
+static struct btp_bap_broadcast_local_source *btp_bap_broadcast_local_source_from_brcst_id_get(
+	uint32_t broadcast_id)
+{
+	for (uint8_t i = 0; i < ARRAY_SIZE(local_sources); i++) {
```

Comment:
General suggestion: Use the same type for the loop variable as the value comparing to
```suggestion
	for (size_t i = 0; i < ARRAY_SIZE(local_sources); i++) {
```

For this specific case it's not a huge deal as `ARRAY_SIZE(local_sources)` is very unlikely to ever become larger than an `uint8_t`, but generally just a good practice :)